modules/examples/destinations/example_destination/example_destination.c

```diff
@@ -93,24 +84,23 @@ _dd_init(LogPipe *d)
   if (!log_threaded_dest_driver_init_method(d))
     return FALSE;
 
-  msg_verbose("Initializing ExampleDestination",
-              evt_tag_str("driver", self->super.super.super.id),
-              evt_tag_str("filename", self->filename),
-              NULL);
+  if (!self->filename)
+    self->filename = g_strdup("/tmp/example-destination-output.txt");
 
-  return log_threaded_dest_driver_start_workers(&self->super);
+  return log_threaded_dest_driver_start_workers(d);
 }
 
-
 gboolean
 _dd_deinit(LogPipe *s)
 {
-  ExampleDestinationDriver *self = (ExampleDestinationDriver *)s;
+  /*
+     If you created resources during init,
+     you need to destroy them here.
 
-  msg_verbose("Deinitializing ExampleDestination",
-              evt_tag_str("driver", self->super.super.super.id),
-              evt_tag_str("filename", self->filename),
-              NULL);
+     self->filename is outside of the lifecycle of init-deinit (may be
```

Comment:
I know this is the common `syslog-ng` way of handling strings like this, but we could simply create a `GString` in `new()`, free it in `free()` and `truncate()` then `set()` it in `init()` or `config-set()`.

Reply:
Added a new patch to the top. Thanks.

Reply:
Thanks!